### PR TITLE
Switched out Promise polyfill due to old one polyfilling chrome

### PR DESCRIPTION
**Pull request by JoshGalvin**
^ Both FF and Chrome will throw uncaught exceptions within promises. Unfortunately the old polyfill library was polyfilling Chrome even though it had native Promise support.

### CODE REVIEW

**Comment by dmarcos** [comment]
Can you remove the dist file from the PR?

**Comment by JoshGalvin** [comment]
Done.

**Comment by JoshGalvin** [comment]
Fixes #796 btw.

File: package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "debug": "^2.2.0",
     "deep-assign": "^2.0.0",
     "document-register-element": "^0.5.2",
-    "es6-promise": "^3.0.2",
+    "promise-polyfill": "^3.1.0",
```
Comment (dmarcos) [inline package.json]: What's the difference between the polyfills?

File: src/index.js [hunk 2]
```diff
@@ -1,4 +1,9 @@
-require('es6-promise').polyfill();  // Polyfill `Promise`.
+// Polyfill `Promise`.
+var promise = require('promise-polyfill');
+if (!global.Promise) {
```
Comment (ngokevin) [inline src/index.js]: maybe ```global.Promise = global.Promise || require('promise-polyfill');```

File: src/index.js [hunk 3]
```diff
@@ -1,4 +1,9 @@
-require('es6-promise').polyfill();  // Polyfill `Promise`.
+// Polyfill `Promise`.
+var promise = require('promise-polyfill');
```
Comment (dmarcos) [inline src/index.js]: Should we do 
````javascript
window.Promise = window.Promise || require('promise-polyfill');
````

Comment (ngokevin) [inline src/index.js]: first! https://github.com/aframevr/aframe/pull/1026#discussion_r54186419

File: src/index.js (hunk 2 above)
Comment (dmarcos) [inline src/index.js]: I don't think that it's even necessary. 
```javascript
require('promise-polyfill');
````
it should be enough. It seems the polyfill already attaches itself to the global scope if Promise is not defined:

https://github.com/taylorhakes/promise-polyfill/blob/master/Promise.js#L196

**Comment by JoshGalvin** [comment]
I prefer using global within the browserify environment as it maps to window and would still work in a node environment.

File: src/index.js [hunk 4]
```diff
@@ -1,4 +1,6 @@
-require('es6-promise').polyfill();  // Polyfill `Promise`.
+// Polyfill `Promise`.
+global.Promise = global.Promise || require('promise-polyfill');
```
Comment (dmarcos) [inline src/index.js]: We should use window instead of global for consistency as we do below in this file.

Comment (JoshGalvin) [inline src/index.js]: Roger, its not like this could ever really run in a node environment anyhow :gun:

**Comment by JoshGalvin** [comment]
@dmarcos the other polyfill was checking "Object.prototype.toString.call(P.resolve()) === '[object Promise]'", which came out as [object Object] in Chrome. Technically you could still use the other library as long as we did the polyfill check ourselves.

**Comment by dmarcos** [comment]
Ok. I had no idea what the difference was.

**Comment by dmarcos** [comment]
Waiting for green!

**Comment by JoshGalvin** [comment]
Its kinda funny because we only need this polyfill to support IE http://caniuse.com/#feat=promises

**Comment by dmarcos** [comment]
AFrame works on IE and almost on windows phone. There's a problem with the values we read from the device orientation API that we will fix on next release.

**Comment by dmarcos** [comment]
Thanks! Congratulations! You're the 8th contributor. I guess we have to order t-shirts soon to give away :). Ideas for a awesome designs are more than welcome.

File: package.json (hunk 1 above)
Comment (cvan) [inline package.json]: I'm also curious. as long as it's A+ compliant, that's legit.

File: src/index.js [hunk 5]
```diff
@@ -1,4 +1,6 @@
-require('es6-promise').polyfill();  // Polyfill `Promise`.
+// Polyfill `Promise`.
+window.Promise = window.Promise || require('promise-polyfill');
```
Comment (cvan) [inline src/index.js]: personally, I check for `'Promise' in window` instead of always assigning

**Pull request merged**